test/Directory.Build.targets

```diff
@@ -0,0 +1,66 @@
+      </RestoreSources>
+      <TestProjectProperties>
+        RestoreSources=$([MSBuild]::Escape($(RestoreSources.Trim())));
+        RuntimeFrameworkVersion=$(RuntimeFrameworkVersion);
+        MicrosoftApplicationInsightsAspNetCorePackageVersion=$(MicrosoftApplicationInsightsAspNetCorePackageVersion);
+        MicrosoftAspNetCoreAuthenticationCookiesPackageVersion=$(MicrosoftAspNetCoreAuthenticationCookiesPackageVersion);
+        MicrosoftAspNetCoreDiagnosticsEntityFrameworkCorePackageVersion=$(MicrosoftAspNetCoreDiagnosticsEntityFrameworkCorePackageVersion);
+        MicrosoftAspNetCoreDiagnosticsPackageVersion=$(MicrosoftAspNetCoreDiagnosticsPackageVersion);
+        MicrosoftAspNetCoreIdentityEntityFrameworkCorePackageVersion=$(MicrosoftAspNetCoreIdentityEntityFrameworkCorePackageVersion);
+        MicrosoftAspNetCoreMvcPackageVersion=$(MicrosoftAspNetCoreMvcPackageVersion);
+        MicrosoftAspNetCorePackageVersion=$(MicrosoftAspNetCorePackageVersion);
+        MicrosoftAspNetCoreServerIISIntegrationPackageVersion=$(MicrosoftAspNetCoreServerIISIntegrationPackageVersion);
+        MicrosoftAspNetCoreServerKestrelPackageVersion=$(MicrosoftAspNetCoreServerKestrelPackageVersion);
+        MicrosoftAspNetCoreStaticFilesPackageVersion=$(MicrosoftAspNetCoreStaticFilesPackageVersion);
+        MicrosoftEntityFrameworkCoreDesignPackageVersion=$(MicrosoftEntityFrameworkCoreDesignPackageVersion);
+        MicrosoftEntityFrameworkCoreSqlServerPackageVersion=$(MicrosoftEntityFrameworkCoreSqlServerPackageVersion);
+        MicrosoftExtensionsConfigurationEnvironmentVariablesPackageVersion=$(MicrosoftExtensionsConfigurationEnvironmentVariablesPackageVersion);
+        MicrosoftExtensionsConfigurationJsonPackageVersion=$(MicrosoftExtensionsConfigurationJsonPackageVersion);
+        MicrosoftExtensionsConfigurationUserSecretsPackageVersion=$(MicrosoftExtensionsConfigurationUserSecretsPackageVersion);
+        MicrosoftExtensionsLoggingConsolePackageVersion=$(MicrosoftExtensionsLoggingConsolePackageVersion);
+        MicrosoftExtensionsLoggingDebugPackageVersion=$(MicrosoftExtensionsLoggingDebugPackageVersion);
+        MicrosoftExtensionsLoggingPackageVersion=$(MicrosoftExtensionsLoggingPackageVersion);
+        MicrosoftExtensionsOptionsConfigurationExtensionsPackageVersion=$(MicrosoftExtensionsOptionsConfigurationExtensionsPackageVersion);
+        MicrosoftVisualStudioWebBrowserLinkPackageVersion=$(MicrosoftVisualStudioWebBrowserLinkPackageVersion);
+        MicrosoftVisualStudioWebCodeGenerationDesignPackageVersion=$(Version);
+        MicrosoftVisualStudioWebCodeGenerationToolsPackageVersion=$(Version);
+      </TestProjectProperties>
+    </PropertyGroup>
+
+    <Sdk_GenerateFileFromTemplate
+      TemplateFile="$(TestProjectTemplateFile)"
+      Properties="$(TestProjectProperties)"
+      OutputPath="$(GeneratedTestProjectFile)">
+      <Output TaskParameter="FileWrites" ItemName="FileWrites" />
```

Comment:
So the compile itemgroup needs to be updated by what's FileWrites used for?

Reply:
FileWrites an MSBuild convention used to keep track of output files produced. The C# compiler, resgen, and other tasks add items to it. It helps when debugging incremental builds.